src/pages/peers/PeerTable.tsx

```diff
@@ -27,28 +27,41 @@ interface Props {
   peers: Peer[] | null
 }
 
+interface PeerLatency {
+  rtt: string
+  loading: boolean
+}
+
+function getPingState(peerLatency: Record<string, PeerLatency>, peer: Peer): ReactElement {
```

Comment:
Hmm, first I found it a bit confusing (why is it not `PingState`?), but I understand this is being used in the `map` function, and that is why you chose this function style?

Reply:
Yeah exactly. I was considering if this should be a component on its own but that seemed quite bizarre. For optimalization purposes it's defined outside of the PeerTable component as this function does not need to be recreated with each re-render.